contracts/superfluid/CFALib.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ISuperToken} from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperfluid.sol";
+import {IConstantFlowAgreementV1} from "@superfluid-finance/ethereum-contracts/contracts/interfaces/agreements/IConstantFlowAgreementV1.sol";
+import {CFAv1Library} from "@superfluid-finance/ethereum-contracts/contracts/apps/CFAv1Library.sol";
+
+library CFALib {
+    using CFAv1Library for CFAv1Library.InitData;
+
+    function _decreaseFlow(
+        CFAv1Library.InitData storage cfaV1,
+        IConstantFlowAgreementV1 _cfa,
+        ISuperToken token,
+        address to,
+        int96 by
+    ) internal {
+        (, int96 curRate, , ) = _cfa.getFlow(token, address(this), to);
+        int96 newRate = curRate - by;
+        require(newRate >= 0, "new rate would be negative");
```

Comment:
This check needs to be moved before line 19.

Reply:
I think you can delete it